src/devtools/client/debugger/src/components/PrimaryPanes/SourcesTreeItem.js

```diff
@@ -30,16 +31,33 @@ import {
 } from "../../utils/source";
 import { isDirectory, getPathWithoutThread } from "../../utils/sources-tree";
 import { copyToTheClipboard } from "../../utils/clipboard";
-import { features } from "../../utils/prefs";
 import { downloadFile } from "../../utils/utils";
 import { isFulfilled } from "../../utils/async-value";
 
 class SourceTreeItem extends Component {
+  _node;
+
   componentDidMount() {
-    const { autoExpand, item } = this.props;
+    const { autoExpand, item, focused } = this.props;
     if (autoExpand) {
       this.props.setExpanded(item, true, false);
     }
+    if (focused) {
+      this._node.parentElement.classList.add("focused");
```

Comment:
reaching outside of the current element is a bit risky... Why do we need to do that here?

Reply:
Very risky.

The SourcesTreeItem is just the tree node. It doesn't include the entire row. Without this. the focus persists when toggling the panel, but the indent doesn't have the focus styling. I added screenshots below.

The proper way to do it would be to apply the focused styling to the entire row. To do that, we'd have to edit code in the `Tree` component. But [we're currently importing it from npm](https://github.com/RecordReplay/devtools/blob/master/src/devtools/client/debugger/src/components/shared/ManagedTree.js#L9), so we can't.

**With parentElement styling:**
![image](https://user-images.githubusercontent.com/15959269/103422206-402b5700-4b6e-11eb-8b31-ba22b4e5f5f3.png)

**Without:**
![image](https://user-images.githubusercontent.com/15959269/103422194-2e49b400-4b6e-11eb-90a9-8b95ee112212.png)

Reply:
ooh... actually 😸 

haven't tested this yet 
https://github.com/RecordReplay/devtools/pull/1439